CMakeLists.txt

```diff
@@ -650,6 +650,20 @@ macro(openssl_check_symbol_exists SYMBOL FILES VARIABLE)
   cmake_pop_check_state()
 endmacro()
 
+# Ensure that the OpenSSL fork actually supports QUIC.
+macro(openssl_check_quic)
+  if(NOT DEFINED HAVE_SSL_CTX_SET_QUIC_METHOD)
+    if(USE_OPENSSL)
+      openssl_check_symbol_exists(SSL_CTX_set_quic_method "openssl/ssl.h" HAVE_SSL_CTX_SET_QUIC_METHOD)
+    elseif(USE_WOLFSSL)
+      openssl_check_symbol_exists(wolfSSL_set_quic_method "wolfssl/options.h;wolfssl/openssl/ssl.h" HAVE_SSL_CTX_SET_QUIC_METHOD)
```

Comment:
do you know whether quiche actually supports wolfssl? afaik it depends on boringssl: https://github.com/cloudflare/quiche#building

so this cmake configure step may succeed with USE_WOLFSSL and USE_QUICHE, but fail during compilation

the USE_QUICHE block should probably include a check for HAVE_BORINGSSL to make sure that openssl_check_quic() actually checked for the correct SSL_CTX_set_quic_method symbol here

Reply:
No, I haven't made tests with quiche yet.

If quiche depends specifically on BoringSSL, it's sensible 
indeed to check for that. Pushed a commit to do this.

Do you know if quiche works with AWS-LC as well? It's Amazon's BoringSSL fork.

Reply:
> No, I haven't made tests with quiche yet.
> 
> If quiche depends specifically on BoringSSL, it's sensible indeed to check for that. Pushed a commit to do this.

thanks, updated branch still works for me 👍 

> 
> Do you know if quiche works with AWS-LC as well? It's Amazon's BoringSSL fork.

i don't see any discussion about AWS-LC on their github. i checked out/built AWS-LC, but when i plugged it into quiche i got undefined references:
```
/path/to/quiche/quiche/src/tls.rs:672: undefined reference to `SSL_get_curve_id'
/usr/bin/ld: /path/to/quiche/quiche/src/tls.rs:677: undefined reference to `SSL_get_curve_name'
```

Reply:
Thanks for testing these.

BoringSSL it is then. We can extend this when quiche adds support for more forks (or vice versa.)

Reply:
who knows, maybe openssl will support quic someday 🙃

Reply:
by _some_ definition of "quic" :)

Reply:
> do you know whether quiche actually supports wolfssl

I'm pretty sure it does not. It is written with BoringSSL in mind.

Reply:
> maybe openssl will support quic someday

They already do, but they implement their own QUIC stack and not the "standard" API for it. curl has no support for OpenSSL's API and AFAIK, neither does nghttp3 so we are quite far off an OpenSSL backend for h3.

Reply:
Yeah, their completely different, newly developed quic API is now in 3.2-alpha.

For extra fun, there was some symbol re-use, but maybe not too bad:
https://github.com/openssl/openssl/issues/22204